public/app/core/services/backend_srv.ts

```diff
@@ -334,15 +324,33 @@ export class BackendSrv implements BackendService {
       message = err.message;
     }
 
+    // Validation
+    if (err.status === 422) {
+      description = err.data.message;
+      message = 'Validation failed';
+    }
+
+    if (config.showErrorAlert === false) {
+      return;
+    }
```

Comment:
Why did we bring this "bail-out-early" down here? (It happened first, because if the condition is true then we don't need assign any variables or evaluate anything else.)

Reply:
Yeah sorry, I reverted my commits but guess something got borked in the process. I'll just bring this back to what it was before as we're not using anything here anymore